### PR TITLE
upgrade eslint to latest version

### PR DESCRIPTION
**Pull request by chenxsan**
I was trying to document [`import.meta.url`](https://github.com/webpack/webpack/pull/11075) introduced in [webpack v5.0.0-beta.21](https://github.com/webpack/webpack/releases/tag/v5.0.0-beta.21) only to find that [the current eslint](https://github.com/webpack/webpack.js.org/blob/master/package.json#L101) (v4.19.1) we have threw error because the `import.meta` feature was added in [eslint v7.2.0](https://github.com/eslint/eslint/pull/13196).

image

This PR should upgrade eslint to the latest version so we can document more advanced JavaScript features that webpack supports.

### CODE REVIEW

**Comment by chenxsan** [comment]
CI failed when updating babel-eslint to v9, related issue https://github.com/babel/babel-eslint/issues/799

**Comment by chenxsan** [comment]
After replacing `babel-eslint` with `@babel/eslint-parser`, [CI failed](https://travis-ci.org/github/webpack/webpack.js.org/jobs/724382750#L266) because we have some typescript typings in JavaScript code block https://github.com/webpack/webpack.js.org/blob/master/src/content/api/module-methods.md#require:

``` javascript
require.resolve(dependency: String);
```

**Comment by chenxsan** [comment]
Now we're able to use `import.meta.url` in JavaScript code block:

image

**Review by evilebottnawi — APPROVED** [review]
Good job!

File: src/content/api/module-methods.md [hunk 1]
```diff
@@ -166,6 +166,8 @@ The goal of CommonJS is to specify an ecosystem for JavaScript outside the brows
 
 ### `require`
 
+<!-- eslint-skip -->
```
Comment (montogeek) [inline src/content/api/module-methods.md:169]: Why is this needed?

Comment (evilebottnawi) [reply src/content/api/module-methods.md:169]: Because it is invalid javascript syntax:
```
require(dependency: String);
```

Comment (EugeneHlushko) [reply src/content/api/module-methods.md:169]: changing ```js to ```ts|typescript works too or not supported?

Comment (chenxsan) [reply src/content/api/module-methods.md:169]: `ts` or `typescript` is another valid option in those cases.

And we do have two occurrences in this project:

image

Do you prefer `ts`/`typescript`?

Comment (EugeneHlushko) [reply src/content/api/module-methods.md:169]: since there's already `typescript` in the codebase, go ahead with that

File: .babelrc [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+    "@babel/preset-env",
+    "@babel/preset-react"
   ],
   "plugins": [
-    "syntax-dynamic-import",
-    "transform-object-rest-spread",
-    "transform-class-properties"
+    "@babel/plugin-syntax-dynamic-import",
+    "@babel/plugin-proposal-object-rest-spread",
+    "@babel/plugin-proposal-class-properties"
```
Comment (evilebottnawi) [inline .babelrc:9]: I think we can remove `plugin-proposal-class-properties`, `plugin-proposal-object-rest-spread` and `plugin-syntax-dynamic-import`, they are in `@babel/preset-env`

Comment (chenxsan) [reply .babelrc:8]: We still need `plugin-proposal-class-properties` and `plugin-syntax-dynamic-import` according to [babel playground](https://babeljs.io/en/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=ATDGBsEMGduAhA9gJwNbAN4CgQgPR4AKyiADgKbIAuAnsAJYB29V9k49AXpcNDY1UgAPHLibRBjUOWJlKtYAF5gAIgBGKVCoDcokBoCujACYAxI6FaJGS4AAoAlEoB8mPbmTkqB5DaoALemgAOnFJaVkKahpdXGAAX113AgBlQVZQMChYYFISKNZyaHcJSAzedPpQSPk6ZXVINXJwAEloAGFERHAdEsrM0ozzKSsbZQAzC1HHNziQT29fBE1gwaqa6PdE0XisegBbUhQqOwByYLxIU4dggPJGO0gXMGtobvJg8EQAc0eHB10oFeVGAalsGG2QMYEmAT2UGGCiLUiSAA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env&prettier=false&targets=&version=7.5.5&externalPlugins=%40babel%2Fplugin-proposal-class-properties%407.10.4%2C%40babel%2Fplugin-syntax-dynamic-import%407.8.3).

Comment (evilebottnawi) [reply .babelrc:8]: https://github.com/babel/babel/blob/main/packages/babel-preset-env/package.json#L17

Comment (chenxsan) [reply .babelrc:8]: Do you mean we can remove these from `devDependencies` or from `.babelrc`?

Comment (evilebottnawi) [reply .babelrc:8]: And from `devDependencies` and from `.babelrc`

Comment (chenxsan) [reply .babelrc]: I've pushed another commit removing all those plugins, as you can see, both the Vercel deployment and [CI failed](https://travis-ci.org/github/webpack/webpack.js.org/jobs/725574269#L255).

Regarding the `devDependencies`, since we're using those plugins in `.babelrc`, wouldn't it be a good idea to define them explicitly instead of using those included in `@babel/preset-env`?

File: .babelrc [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
-    "env",
-    "react"
+    "@babel/preset-env",
+    "@babel/preset-react"
   ],
   "plugins": [
-    "syntax-dynamic-import",
-    "transform-object-rest-spread",
-    "transform-class-properties"
+
```
Comment (evilebottnawi) [inline .babelrc:7]: Empty section :smile:

Comment (chenxsan) [reply .babelrc]: I've removed the whole `plugins` property (hope I didn't misunderstand you again), still failed :(

Comment (evilebottnawi) [reply .babelrc]: Very strange, `babel-present-env` contains `@babel/plugin-proposal-class-properties`, so we don't need it in our plugins, but looks it was not released right now :confused: , so let's return `@babel/plugin-proposal-class-properties` in plugins array

Comment (chenxsan) [reply .babelrc]: Done.

**Review by evilebottnawi — APPROVED** [review]
Good job!

**Review by EugeneHlushko — APPROVED** [review]
LGTM, nice to see the eslint ignore comments gone :)

**Comment by EugeneHlushko** [comment]
Thanks!
